conjure/servicewriter.go

```diff
@@ -810,6 +810,40 @@ func serviceStructMethodBodyAST(endpointDefinition spec.EndpointDefinition, retu
 	body = append(body, ifErrNotNilReturnHelper(hasReturnVal, valVarToReturnInErr, errVar, nil))
 
 	if returnsBinary {
+		isOptional, err := isReturnTypeSpecificType(endpointDefinition.Returns, visitors.IsOptional)
+		if err != nil {
+			return nil, err
+		}
+		if isOptional {
+			// If an endpoint with a return type of optional<binary> provides a response with a code of StatusNoContent
+			// then the return value is empty and nil is returned.
+			body = append(body, &statement.If{
+				Cond: &expression.Binary{
+					LHS: expression.NewSelector(
+						expression.VariableVal(respVar),
+						"StatusCode",
+					),
+					Op:  token.EQL,
+					RHS: expression.Type("http.StatusNoContent"),
+				},
+				Body: []astgen.ASTStmt{
+					statement.NewReturn(
+						expression.Nil,
+						expression.Nil,
+					),
+				},
+			})
+			info.AddImports("net/http")
+			// if endpoint returns binary, return pointer to body of response directly
+			body = append(body, statement.NewReturn(
+				expression.NewSelector(
+					expression.VariableVal("&"+respVar),
```

Comment:
Sorry, forgot to comment on this in last review: this `body = append(body, statement.NewReturn(...` block is the same as the block outside the "if" statement except for the variable value being `"&"+respVar`. Can we unify that logic and return to simplify branching? Can either create variable before the `if isOptional` and assign accordingly or do the `if isOptional` check after to create the variable val before the `body` assignment.